src/Http/Http.Results/src/PhysicalFileResult.cs

```diff
@@ -105,6 +105,14 @@ public Task ExecuteAsync(HttpContext httpContext)
         public FileInfoWrapper(string path)
         {
             var fileInfo = new FileInfo(path);
+
+            // It means we are dealing with a symlink and need to get the information
+            // from the target file instead.
+            if (fileInfo.Exists && fileInfo.ResolveLinkTarget(true) is FileInfo linkTargetInfo)
```

Comment:
@adamsitnik / @eiriktsarpalis do you happen to know if this is expected? According to https://github.com/dotnet/aspnetcore/issues/39170, `FileInfo.Length` isn't correct for symbolic links and you have to resolve the link to get the right value. If it is, should we document this behavior as a remark in https://docs.microsoft.com/dotnet/api/system.io.fileinfo.length?view=net-6.0#System_IO_FileInfo_Length?

Reply:
@pranavkm I will let @adamsitnik / @eiriktsarpalis  give their position but  based on this PR https://github.com/dotnet/corefx/pull/6591 (from @stephentoub back in 2016) the `FileInfo.Length` is expected to not return the correct value for symbolic links, in both Windows and Unix.

![image](https://user-images.githubusercontent.com/12376385/148295555-c897a346-b0c9-4ef0-8c66-36273c1574e8.png)

Also, this is a bug similarly reported in the past  https://github.com/dotnet/aspnetcore/issues/2774, however I did not check if this scenario is fixed with my change.